src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -455,43 +457,125 @@ def create_resource(properties,
     return res.create_resource(properties, location, is_full_object)
 
 
-def show_resource(resource_group_name=None,
+def show_resource(resource_ids=None, resource_group_name=None,
                   resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
                   resource_name=None, api_version=None):
+    if resource_ids:
+        results = []
+        for rid in resource_ids:
+            parsed_id = parse_resource_id(rid)
+            if len(parsed_id) == 1:
```

Comment:
Please clarify: why does a one-character-long parsed ID mean the id is invalid? Is it possible a two-characters-long ID presents a failure?

Reply:
It the type of `parsed_id` is not a string, it worths to point out here.